### PR TITLE
Fix address pool panics on start up

### PR DESCRIPTION
**Pull request by cjepson**
Add safety checks for address pool access
    
    Address pool access now has checks to ensure that the address
    pool exists and has been initialized before calls accessing its
    functions or internals.

### CODE REVIEW

File: wallet/addresspool.go [hunk 1]
```diff
@@ -267,9 +267,42 @@ func (w *Wallet) CloseAddressPools() {
 	return
 }
 
+// CheckAddressPoolsInitialized checks to make sure an address pool exists
+// that that one can safely access functions and internal memory such as
+// mutexes.
+func (w *Wallet) CheckAddressPoolsInitialized(account uint32) error {
+	if w.addrPools[waddrmgr.DefaultAccountNum] == nil {
+		return fmt.Errorf("Address pools for account %v"+
```
Comment (alexlyp) [inline wallet/addresspool.go]: need an extra whitespace here

File: wallet/addresspool.go [hunk 2]
```diff
@@ -267,9 +267,42 @@ func (w *Wallet) CloseAddressPools() {
 	return
 }
 
+// CheckAddressPoolsInitialized checks to make sure an address pool exists
+// that that one can safely access functions and internal memory such as
+// mutexes.
+func (w *Wallet) CheckAddressPoolsInitialized(account uint32) error {
+	if w.addrPools[waddrmgr.DefaultAccountNum] == nil {
```
Comment (alexlyp) [inline wallet/addresspool.go]: shouldn't this be account not waddrmgr.DefaultAccountNum ?

**Comment by alexlyp** [comment]
looks like CheckAddressPoolsInitialized should be using account not DefaultAccoutnNum

**Comment by jcvernaleo** [comment]
Other than the missing space @alexlyp mentioned, this works for me too.
tACK

**Comment by cjepson** [comment]
Ok, update

**Comment by jcvernaleo** [comment]
OK this one is good to go too.

**Pull request merged**